src/prefect/client/schemas/objects.py

```diff
@@ -305,6 +305,9 @@ def is_cancelled(self) -> bool:
     def is_cancelling(self) -> bool:
         return self.type == StateType.CANCELLING
 
+    def is_cached(self) -> bool:
+        return self.name == "Cached"
```

Comment:
nit: but i think we'd want to check the `type` _and_ `name` here since technically anyone could call any state `type` any `name`